wordpress-mobile/WordPress-iOS
# fix: Preview adjusts for device orientation change

Fixes #18483. After orientation changes, the device width may be too narrow for the existing preview, which causes it to overflow the view. This overflow is more prominent when Display Zoom is enabled, but is present regardless. By resetting the preview width after orientation change, we ensure the preview fits the view. 

<details><summary>Before</summary>

https://user-images.githubusercontent.com/438664/175136889-8e4b3c83-d266-4d49-b15b-9ec5c25353f1.mov

</details>

<details><summary>After</summary>

https://user-images.githubusercontent.com/438664/175137277-42547889-c3a7-4872-83ba-5f55c1d83832.mov

</details>

To test:

> **Note**: It is important to test on both an iPhone and iPad, as the reporting of view dimensions seems to vary between the device types. 

<details><summary>While previewing, rotate from portrait to landscape orientation</summary>

1. Open a published or drafted post.   
1. Position your device in _portrait_ orientation. 
1. Tap the three-dot menu in the top-right of the editor, select _Preview_.
1. Tap the device icon in the bottom-right and select a preview device size.
1. Rotate your device to _landscape_ orientation. 
1. Verify the preview does not overflow the width of the device; the entirety of the preview is visible on the screen.<sup>1</sup> 
1. Rotate your device to _portrait_ orientation. 
1. Verify the preview does not overflow the width of the device; the entirety of the preview is visible on the screen.<sup>1</sup> 
1. Repeat steps 4-8 with each of the preview device sizes available. 

<sup>1</sup> One way to verify the correct display is to again toggle the device away and back to your current preview device, e.g. "mobile" → "desktop" → "mobile" without rotating the device. This will re-render the preview and provide a clear picture as to what the preview should look like. 

</details>

<details><summary>While previewing, rotate from landscape to portrait orientation</summary>

1. Open a published or drafted post.   
1. Position your device in _landscape_ orientation. 
1. Tap the three-dot menu in the top-right of the editor, select _Preview_.
1. Tap the device icon in the bottom-right and select a preview device size.
1. Rotate your device to _portrait_ orientation. 
1. Verify the preview does not overflow the width of the device; the entirety of the preview is visible on the screen.<sup>1</sup> 
1. Rotate your device to _landscape_ orientation. 
1. Verify the preview does not overflow the width of the device; the entirety of the preview is visible on the screen.<sup>1</sup> 
1. Repeat steps 4-8 with each of the preview device sizes available. 

<sup>1</sup> One way to verify the correct display is to again toggle the device away and back to your current preview device, e.g. "mobile" → "desktop" → "mobile" without rotating the device. This will re-render the preview and provide a clear picture as to what the preview should look like. 

</details>

## Regression Notes
1. Potential unintended areas of impact
  The preview web view could experience bugs during rotation or memory leaks from the changes.
2. What I did to test those areas of impact (or what existing automated tests I relied on)
  Manually tested various devices, zoom settings, and multiple rotation events.
3. What automated tests I added (or what prevented me from doing so)
  None, I am interested in recommendations/guidance on if/what automated testing should be added.

PR submission checklist:
